src/cdk-experimental/testing/protractor/protractor-element.ts

```diff
@@ -55,4 +55,8 @@ export class ProtractorElement implements TestElement {
     const classes = (await this.getAttribute('class')) || '';
     return new Set(classes.split(/\s+/).filter(c => c)).has(name);
   }
+
+  async getPropertyValue(name: string): Promise<any> {
```

Comment:
Why is this necessary? `getAttributeValue` should fall back to getting the property

Reply:
This is what I had as commit description when I did the same changes on another PR:

```
Most of the time harnesses are able to use `getAttribute` to retrieve data
from DOM elements, but there are edge-cases where it's not correct to
retrieve data through the attribute.

For example: Consider an `input` element which has a `value` attribute set to
the initial input value. Now when the value of the input changes through
interaction, the `value` attribute is not refreshed as it's a static attribute.

In that case, the harness should retrieve the value by reading the DOM element
property. This is needed in order to implement the `MatInputHarness`.
```

Reply:
Ah I see, does it still make sense to have `getAttribute` fallback to getting the property then? Considering that we have a separate method for getting the property now. 

The decision to have the fallback behavior was part of the original internal harness framework, I'm assuming because that's how protractor's getAttribute method works.

@jelbourn do you have an opinion?

Reply:
@mmalerba Yeah IIRC protractor by default also falls back to the property value. I could see the point of `getAttribute` only checking the attribute though. 

FWIW through, it was in some cases convenient that it falls back to the property, but we could implement that on our own if needed.

Reply:
Yeah I can see why we might want the fallback behavior. In most cases users probably don't care which it is, `value` is kind of a special case

Reply:
I would lean towards special-casing attributes where we know this doesn't work, but we can discuss at the next team meeting

Reply:
By "special-casing" do you mean preferring the property over the attribute in the case of properties like `value`? If so, is there a chance that could prevent a user from accessing the `value` attribute when that really is what they want to do?